cmd/snap-bootstrap/bootstrap/bootstrap.go

```diff
@@ -29,19 +29,42 @@ type Options struct {
 	// will contain encryption later
 }
 
+func deviceFromRole(lv *gadget.LaidOutVolume, role string) (device string, err error) {
+	for _, vs := range lv.LaidOutStructure {
+		// XXX: this part of the finding maybe should be a
+		// method on gadget.*Volume
+		if vs.Role == role {
```

Comment:
Being somewhat sleepy still, we validate that there's only one instance of each role, right?

Reply:
We do have a check for this but I am not sure we actually have a unittest for this, I will add one.